service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -196,11 +196,16 @@ public void updateMetadata(DatasetId datasetId, String metadata) {
   }
 
   public DatasetId createDataset(
-      StorageSystem storageSystem, String storageSourceId, String metadata) {
-    jsonValidationService.validateMetadata(toJsonNode(metadata));
-    var dataset = new Dataset(storageSourceId, storageSystem, metadata);
-    ensureActionPermission(dataset, SamAction.CREATE_METADATA);
-    return datasetDao.create(dataset).id();
+      StorageSystem storageSystem, String storageSourceId, Map<String, Object> metadata) {
+    try {
+      String metadataAsString = objectMapper.writeValueAsString(metadata);
```

Comment:
I think this should be a service level change, because mapping to a string for internal storage is business logic. My belief is that controller code should purely be used for HTTP validation.